osu.Game/Screens/Play/HUD/FailingLayer.cs

```diff
@@ -73,16 +75,29 @@ private void load(OsuColour color, OsuConfigManager config)
         {
             boxes.Colour = color.Red;
 
-            enabled = config.GetBindable<bool>(OsuSetting.FadePlayfieldWhenHealthLow);
+            configEnabled = config.GetBindable<bool>(OsuSetting.FadePlayfieldWhenHealthLow);
             enabled.BindValueChanged(e => this.FadeTo(e.NewValue ? 1 : 0, fade_time, Easing.OutQuint), true);
+
+            updateBindings();
         }
 
         public override void BindHealthProcessor(HealthProcessor processor)
         {
             base.BindHealthProcessor(processor);
 
-            // don't display ever if the ruleset is not using a draining health display.
-            if (!(processor is DrainingHealthProcessor))
+            healthProcessor = processor;
+            updateBindings();
+        }
+
+        private void updateBindings()
+        {
+            if (configEnabled == null || healthProcessor == null)
```

Comment:
can we make this a `LoadState >= Ready` check instead? took me a bit of mental effort to ensure this wouldn't bind more than once.

Reply:
Just to confirm, you want the initial `updateBindings()` call that's currently inside `load()` to also be moved to `LoadComplete()`?

Reply:
nah, fine where it is, but just would rather than conditional check here is on LoadState rather than null checks (i think it will mean the same thing at the end of the day, right?)

Reply:
It's not fine where it is, that condition will exactly break the scenario I intended to fix here - the potential of calling `BindHealthProcessor()` before `load()`.

Reply:
`if (LoadState < Ready) return` is what we usually use in these cases, along with a call from load or loadcomplete (either is fine).